vividus-plugin-azure-service-bus/src/main/java/org/vividus/azure/servicebus/service/ServiceBusService.java

```
@@ -17,57 +17,153 @@
+            LOGGER.info("The message was successfully sent to {} in {} namespace", name, namespace);
         }
         catch (ServiceBusException e)
         {
             throw new ServiceBusSendMessageException("Unable to send message to Azure Service Bus", e);
         }
     }
+
+    public void startConsuming(String clientKey)
+    {
+        BlockingQueue<ServiceBusReceivedMessage> messages = new LinkedBlockingDeque<>();
+        testContext.get(MESSAGES_KEY, HashMap::new).put(clientKey, messages);
+        ServiceBusClientEntity serviceBusClientEntity = getServiceBusClientConfig(clientKey);
+
+        String namespace = serviceBusClientEntity.getNamespace();
+        ServiceBusClientBuilder.ServiceBusProcessorClientBuilder processorBuilder = clientBuilder
+                .fullyQualifiedNamespace(namespace + DEFAULT_NAMESPACE_POSTFIX)
+                .processor()
+                .processMessage(context -> {
+                    ServiceBusReceivedMessage message = context.getMessage();
+                    BinaryData messageBody = message.getBody();
+                    LOGGER.info("Processing message. Session: {}, Sequence #: {}. Contents: {}", message.getMessageId(),
+                            message.getSequenceNumber(), messageBody);
+                    messages.add(message);
+                })
+                .processError(context ->
+                {
+                    throw (ServiceBusException) context.getException();
+                });
+        ChannelType channelType = serviceBusClientEntity.getChannelType();
+        String name = serviceBusClientEntity.getName();
+        if (channelType == ChannelType.QUEUE)
+        {
+            processorBuilder.queueName(name);
+        }
+        else
+        {
+            processorBuilder.topicName(name);
+            processorBuilder.subscriptionName(serviceBusClientEntity.getSubscriptionName());
```

we need to throw an error if subscription name is configured for queue